toml/src/main/kotlin/org/rust/toml/inspections/NewCrateVersionAvailableInspection.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.inspections
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import org.rust.toml.crates.local.CargoRegistryCrate
+import org.rust.toml.crates.local.CrateVersionRequirement
+import org.rust.toml.stringValue
+import org.toml.lang.psi.TomlValue
+
+class NewCrateVersionAvailableInspection : CrateVersionInspection() {
+    override fun handleCrateVersion(
+        dependency: DependencyCrate,
+        crate: CargoRegistryCrate,
+        versionElement: TomlValue,
+        holder: ProblemsHolder
+    ) {
+        val versionText = versionElement.stringValue ?: return
+        val versionReq = CrateVersionRequirement.build(versionText) ?: return
+        if (versionReq.isPinned) return
+
+        val versions = crate.versions
+            .filter { !it.isYanked }
+            .mapNotNull { it.semanticVersion }
+            .sortedDescending()
+
+        val highestMatchingVersion = versions.firstOrNull { versionReq.matches(it) } ?: return
+        val newerVersion = versions.firstOrNull { it.isStable && it > highestMatchingVersion } ?: return
```

Comment:
What if we will offer to update versions before `1.0` if there are only nightly versions available? It can be useful for crates that are non-stable for a long time

Reply:
It's true that Rust crate authors have a certain.. "reluctancy" to go above 0.x :D But I'm not sure if offering nightly or (semver) unstable 0.xyz versions.

Do you propose to include a special case and also offer newer versions if they are 0.x.y.z (i.e. unstable by semver, but "stable" by Rust standards)?

Reply:
I think it will be good to have such a special case. We have `weak warning` severity for this inspection, so it shouldn't be that annoying to the users

Reply:
Ok, I added support for versions with major version 0.